src/libs/ajax/GoogleStorage.ts

```diff
@@ -1,15 +1,58 @@
 import _ from 'lodash/fp'
 import * as qs from 'qs'
-import { fetchBuckets, saToken } from 'src/libs/ajax'
-import { authOpts, fetchOk, jsonBody } from 'src/libs/ajax/ajax-common'
+import { saToken } from 'src/libs/ajax'
+import { authOpts, fetchOk, jsonBody, withRetryOnError, withUrlPrefix } from 'src/libs/ajax/ajax-common'
+import { withRequesterPays } from 'src/libs/ajax/Billing'
 import { getConfig } from 'src/libs/config'
 import * as Utils from 'src/libs/utils'
-import { getExtension, getFileName, tools } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
-
+import {
+  AbsolutePath,
+  AnalysisFile,
+  FileMetadata,
+  getDisplayName,
+  getExtension,
+  getFileName
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import { getToolFromFileExtension, ToolLabel, tools } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { cloudProviderTypes } from 'src/pages/workspaces/workspace/workspace-utils'
+
+// requesterPaysError may be set on responses from requests to the GCS API that are wrapped in withRequesterPays.
+// requesterPaysError is true if the request requires a user project for billing the request to. Such errors
+// are not transient and the request should not be retried.
+export const fetchBuckets = _.flow(withRequesterPays, withRetryOnError(error => Boolean(error.requesterPaysError)), withUrlPrefix('https://storage.googleapis.com/'))(fetchOk)
 
 const encodeAnalysisName = name => encodeURIComponent(`notebooks/${name}`)
 
+interface GoogleFileRaw {
+  name: string,
+  updated: string,
+  metadata: FileMetadata
+}
+
+interface GoogleStorageListRaw {
+  items: GoogleFileRaw[]
+}
```

Comment:
Is this abstraction necessary vs. the following call site without it?

```typescript
const items = await res.json() as GoogleFileRaw[]
```

Reply:
yes, as `items` is a field in the actual json return, in which the files are (among some other top-level data we don't care abuot)

Reply:
#3468 adds types for `GoogleStorage.list`.